build-system/tasks/extension-generator/index.js

```diff
@@ -237,46 +238,37 @@ async function makeAmpExtension() {
     log(red('Error! Please pass in the "--name" flag with a value'));
   }
   const {name, version = '0.1'} = argv;
+
   const examplesFile = getExamplesFile(name);
 
-  fs.mkdirpSync(`extensions/${name}/${version}/test`);
-  fs.writeFileSync(
-    `extensions/${name}/${name}.md`,
-    await getMarkdownDocFile(name)
-  );
-  fs.writeFileSync(
-    `extensions/${name}/validator-${name}.protoascii`,
-    getValidatorFile(name)
-  );
-  fs.writeFileSync(
-    `extensions/${name}/${version}/${name}.js`,
-    getJsExtensionFile(name)
-  );
-  fs.writeFileSync(
-    `extensions/${name}/${version}/${name}.css`,
-    await getAmpCssFile(name)
-  );
-  fs.writeFileSync(
-    `extensions/${name}/${version}/test/test-${name}.js`,
-    getJsTestExtensionFile(name)
-  );
-  fs.writeFileSync(
-    `extensions/${name}/${version}/test/validator-${name}.html`,
+  const examplesFileValidatorOut =
+    'PASS\n' +
     examplesFile
-  );
-
-  const examplesFileValidatorOut = examplesFile
-    .trim()
-    .split('\n')
-    .map((line) => `|  ${line}`)
-    .join('\n');
+      .trim()
+      .split('\n')
+      .map((line) => `|  ${line}`)
+      .join('\n');
+
+  const fileContent = {
+    [`extensions/${name}/${name}.md`]: getMarkdownDocFile(name),
+    [`extensions/${name}/validator-${name}.protoascii`]: getValidatorFile(name),
+    [`extensions/${name}/${version}/${name}.js`]: getJsExtensionFile(name),
+    [`extensions/${name}/${version}/${name}.css`]: getAmpCssFile(name),
```

Comment:
I think `getMarkdownDocFile` and `getAmpCssFile` are the only async ones here. Maybe consider either modifying those to use `readFileSync` or moving the `await` to the call sites here, and removing it from the for loop below to make it clearer where async things are happening

More generally we could potentially queue up all the reads/writes and `await Promise.all`, but I'd guess that wouldn't be meaningfully better than doing everything synchronously for a task as small as this.

Reply:
I think I prefer the existing pattern because it doesn't matter whether the generator function is async or not.